README.md

```diff
@@ -4,3 +4,48 @@
 
 1. Download Go 1.17 from [here](https://go.dev/doc/install).
 2. Download `golangci-lint` from [here](https://golangci-lint.run/usage/install/).
+
+## Contribution Guidelines
+A lot of these guidelines are from the [SOMAS2020 repo](https://github.com/SOMAS2020/SOMAS2020/blob/main/docs/SETUP.md) :)
+### Coding Rules
+1. You're encouraged to use [VSCode](https://code.visualstudio.com/) with the [Go extension](https://code.visualstudio.com/docs/languages/go).
+2. Trust the language server. Red lines == death. Yellow lines == close to death. An example where it might be very tempting to let yellow lines pass are in `struct`s:
+```golang
+type S struct {
+    name string
+    age  int
+}
+
+s1 := S {"pitt", 42} // NO
+s2 := S {"pitt"} // NO (even though it initialises age to 0)
+
+s3 := S{name: "pitt", age: 42} // OK, if we add fields into S, this will still be correct
+s4 := S{name: "pittson"} // OK if `pittson`'s age is 0
+```
+3. Write tests where required. There is ample guide online on how to do this in Golang. Tests will be run alongside CI when you pull into the main repo. If anyone breaks anything, it's easy to observe that if you have tests. Otherwise, your code will be broken unknowingly.
```

Comment:
```suggestion
3. Write tests where required. There are many guides online on how to do this in Golang. Tests will be run alongside CI when you pull into the main repo. If anyone breaks anything, it's easy to observe that if you have tests. Otherwise, your code will be broken unknowingly.
```